src/raw/serde/bson_visitor.rs

```diff
@@ -44,6 +44,160 @@ use super::{
 /// A visitor used to deserialize types backed by raw BSON.
 pub(crate) struct OwnedOrBorrowedRawBsonVisitor;
 
+pub(super) enum MapParse<'de> {
```

Comment:
This is a little awkward; I was hoping to be able to simply delegate via calling `visit_map`, but the need to signal the aggregate case with the first key included means it didn't fit into that signature.